### PR TITLE
Add autoscaler-keda configmap

### LINKED ISSUE #21: Support a config map for configuring the component
Global configuration options such as `default prometheus address` and other specific settings should be tunable via a config map.

### PR DESCRIPTION
**Pull request by skonto**
# Changes

- Adds a dedicated cm to configure global properties.
- Tested locally by changing the prometheus address and then checking if the related scaledObject spec is updated.
- Cleans up cm deps. For now wrt changes we care about the Serving autoscaler cm and the autoscaler keda one.
- Fixes the scaledobject update logic in the reconciler, previously this would fail due to not properly copying before updating.
 

Fixes #21

### CODE REVIEW

File: pkg/reconciler/autoscaling/hpa/config/testdata/config-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,209 @@
+# Copyright 2024 The Knative Authors
```
Comment (ReToCode) [inline pkg/reconciler/autoscaling/hpa/config/testdata/config-autoscaler.yaml:1]: is this file the correct one or a left over from copying?

Comment (ReToCode) [reply pkg/reconciler/autoscaling/hpa/config/testdata/config-autoscaler.yaml:1]: Or is it needed?

Comment (skonto) [reply pkg/reconciler/autoscaling/hpa/config/testdata/config-autoscaler.yaml:1]: The new config store has two cms that is watching: a) the serving autoscaler one, b) the cm for this component.
So we need this in tests https://github.com/knative-extensions/autoscaler-keda/blob/231941697d82aaff0bceeea95f225b9b01dd917f/pkg/reconciler/autoscaling/hpa/config/store_test.go#L33.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**